### PR TITLE
neomutt: update to 20231023

### PR DESCRIPTION
**Pull request by AnInternetTroll**
#### Testing the changes
- I tested the changes in this PR: **briefly**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64-glibc)

### CODE REVIEW

**Comment by AnInternetTroll** [comment]
@Vaelatern could you take a look?

**Comment by Vaelatern** [comment]
Adding a custom patch? That seems weird -- yes it's committed to upstream but I think I'd rather stay on 10-23 than carry the patch if needed to fix 11-03

File: srcpkgs/neomutt/template [hunk 1]
```diff
@@ -19,11 +19,12 @@ short_desc="Mutt with misc fixes and feature patches"
 maintainer="Toyam Cox <Vaelatern@voidlinux.org>"
 license="GPL-2.0-or-later"
 homepage="https://neomutt.org/"
-_test_files_hash=34f79ee9d91863a97f6e18700aea51157b8610cf
+_test_files_hash=00efc8388110208e77e6ed9d8294dfc333753d54
```
Comment (Vaelatern) [inline srcpkgs/neomutt/template:22]: I mean I guess you could do this... the reason I'd ignored it is the latest commit is a documentation only update. Not actually functionally required.

Comment (Vaelatern) [reply srcpkgs/neomutt/template:22]: But I'll accept this.

**Comment by Vaelatern** [comment]
Can you please give clarity on that patch, which seems to fix a bug in 1103 that we could avoid by being on 1023 without a real loss of functionality, yes?

**Comment by AnInternetTroll** [comment]
On Thu, Nov 16, 2023 at 06:34:57AM -0800, Toyam Cox wrote:
>@Vaelatern commented on this pull request.
>> @@ -19,11 +19,12 @@ short_desc="Mutt with misc fixes and feature patches"
> maintainer="Toyam Cox ***@***.***>"
> license="GPL-2.0-or-later"
> homepage="https://neomutt.org/"
>-_test_files_hash=34f79ee9d91863a97f6e18700aea51157b8610cf
>+_test_files_hash=00efc8388110208e77e6ed9d8294dfc333753d54
>
>I mean I guess you could do this... the reason I'd ignored it is the latest commit is a documentation only update. Not actually functionally required.
Opsy, didn't really check the commit. Just wanted to make sure the
newest tests passed :D
I think I'll keep it just to not cause confusion in the future

**Comment by AnInternetTroll** [comment]
On Sat, Nov 18, 2023 at 10:48:55PM -0800, Toyam Cox wrote:
>Can you please give clarity on that patch, which seems to fix a bug in 1103 that we could avoid by being on 1023 without a real loss of functionality, yes?
Well 1103 does introduce like 4 new features. But it might be safer to
stay on 1023 until that patch comes in a full release.

Just tested 1023 and it doesn't require any patches to build and test
locally, so that seems like the best option.

**Comment by Vaelatern** [comment]
Thank you!

**Pull request merged**